### PR TITLE
zig 0.12.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

---

- https://ziglang.org/download/0.12.0/release-notes.html

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  ld: warning: dylib (/opt/homebrew/opt/llvm@17/lib/libunwind.dylib) was built for newer macOS version (12.7) than being linked (12.0)
  Undefined symbols for architecture arm64:
    "std::__1::__libcpp_verbose_abort(char const*, ...)", referenced from:
        std::__1::__throw_out_of_range[abi:un170006](char const*) in libzigcpp.a(zig_clang_driver.cpp.o)
        std::__1::__throw_length_error[abi:un170006](char const*) in libzigcpp.a(zig_clang_driver.cpp.o)
        std::__1::__throw_length_error[abi:un170006](char const*) in libzigcpp.a(zig_llvm.cpp.o)
        std::__throw_bad_array_new_length[abi:un170006]() in libzigcpp.a(zig_llvm.cpp.o)
        std::__1::__throw_length_error[abi:un170006](char const*) in libzigcpp.a(zig_llvm-ar.cpp.o)
        std::__throw_bad_array_new_length[abi:un170006]() in libzigcpp.a(zig_llvm-ar.cpp.o)
        std::__1::__throw_length_error[abi:un170006](char const*) in libzigcpp.a(zig_clang_cc1as_main.cpp.o)
        ...
  ld: symbol(s) not found for architecture arm64
```

**Comment by softprops** [comment]
nice to see this moving along. in the meantime I found `asdf install zig 0.12.0` works out of the box

File: .github/autobump.txt [hunk 1]
```diff
@@ -2187,6 +2187,7 @@ zellij
 zenith
 zeromq
 zf
+zig
```
Comment (carlocab) [inline .github/autobump.txt:2190]: Zig isn't really a good candidate for autobump, because it usually requires multiple changes during version bumps

Comment (chenrui333) [reply .github/autobump.txt:2190]: i think we can just get the bump PR (which is automated), and put changes in the PR.

**Comment by chenrui333** [comment]
> ```
>   ld: warning: dylib (/opt/homebrew/opt/llvm@17/lib/libunwind.dylib) was built for newer macOS version (12.7) than being linked (12.0)
>   Undefined symbols for architecture arm64:
>     "std::__1::__libcpp_verbose_abort(char const*, ...)", referenced from:
>         std::__1::__throw_out_of_range[abi:un170006](char const*) in libzigcpp.a(zig_clang_driver.cpp.o)
>         std::__1::__throw_length_error[abi:un170006](char const*) in libzigcpp.a(zig_clang_driver.cpp.o)
>         std::__1::__throw_length_error[abi:un170006](char const*) in libzigcpp.a(zig_llvm.cpp.o)
>         std::__throw_bad_array_new_length[abi:un170006]() in libzigcpp.a(zig_llvm.cpp.o)
>         std::__1::__throw_length_error[abi:un170006](char const*) in libzigcpp.a(zig_llvm-ar.cpp.o)
>         std::__throw_bad_array_new_length[abi:un170006]() in libzigcpp.a(zig_llvm-ar.cpp.o)
>         std::__1::__throw_length_error[abi:un170006](char const*) in libzigcpp.a(zig_clang_cc1as_main.cpp.o)
>         ...
>   ld: symbol(s) not found for architecture arm64
> ```

probably related, https://github.com/llvm/llvm-project/issues/87590

**Review by fxcoudert — DISMISSED** [review]

File: .github/autobump.txt (hunk 1 above)
Comment (carlocab) [reply .github/autobump.txt:2208]: That means contributors can't use `brew bump[-formula-pr]` to work on an update themselves though

**Review by p-linnane — APPROVED** [review]

**Pull request merged**